pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -712,6 +753,20 @@
       %^  return-static-data-on-duct  404  'text/html'
       (error-page 404 authenticated url.request ~)
     ==
+    :: test if host header is valid and turf in domains.state
+    ::
+    ++  host-in-domains
```

Comment:
You may be able to deduplicate this with the existing logic in `+get-action-for-binding` and `+host-matches`. Not immediately obvious what the elaborate `+host-parser` is for, isn't the host header always just `host(:port)`?

Reply:
the host-parser is mostly an overly RFC compliant parser for the userinfo portion eg `foo:bar@[host][port]`. The zuse url parsing function that handles it just uses an @ta parser which is inadequate as it doesn't handle uppercase letters and other legal symbols.

The basic issue is I have to concat 'https://' + host + path. When I tested with curl I found the runtime will pass through any insane value for the host header like  `!!!  \\  e\\\\` or w/e without restriction which will result in an invalid location header for the redirect. I wanted to err on the side of only producing a valid redirect location but maybe that's not necessary. Arguably it's still too restrictive as I think the zuse domain parser doesn't handle ipv6 and other weird possibilities.

What you think? Should I just accept any value in the host header and concat it into the location header without caring or should I try simplify the parser by excluding some of the technically legal but unlikely userinfo characters or something else

Reply:
My understanding is that userinfo is always excluded from the host header, as per [the http1.1 spec](https://httpwg.org/specs/rfc7230.html#header.host). Is there something I'm misunderstanding?

My inclination would be to just pass the host into redirect field without even inspecting it. If they send bogus, they receive bogus. If we want to reject requests with strange host headers that should probably happen higher up...

Reply:
oh you're right I didn't realise userinfo was meant to go in a separate header. Yeah I'll just remove that whole check